poc/vdaf_poc/flp_bbcggi19.py

```diff
@@ -902,32 +902,33 @@ def test_vec_set_type_param(self, test_vec: dict[str, Any]) -> list[str]:
 class SumVec(Valid[list[int], list[int], F]):
     EVAL_OUTPUT_LEN = 1
     length: int
-    bits: int
+    max_measurement: int
     chunk_length: int
     field: type[F]
 
     def __init__(self,
                  field: type[F],
                  length: int,
-                 bits: int,
+                 max_measurement: int,
                  chunk_length: int):
         """
         Instantiate the `SumVec` circuit for measurements with
-        `length` elements, each in the range `[0, 2**bits)`.
+        `length` elements, each in the range `[0, max_measurement]`.
         """
         # REMOVE ME
-        if 2 ** bits >= field.MODULUS:
-            raise ValueError('bit size exceeds field modulus')
-        if bits <= 0:
-            raise ValueError('invalid bits')
         if length <= 0:
             raise ValueError('invalid length')
         if chunk_length <= 0:
             raise ValueError('invalid chunk_length')
+        if max_measurement <= 0:
+            raise ValueError('invalid max_measurement')
 
         self.field = field
         self.length = length
+        bits = max_measurement.bit_length()
         self.bits = bits
+        self.max_measurement = max_measurement
+        self.last_weight = max_measurement - (2 ** (bits - 1) - 1)
```

Comment:
nit: It might be nice to give this a name since we reuse it. "mask" may not be right.

```suggestion
        self.last_mask = (2 ** (bits - 1) - 1)
        self.last_weight = max_measurement - self.last_mask
```